### PR TITLE
A fix and a feaure

### PR DESCRIPTION
**Pull request by alexgleith**
Add a `--dryrun` flag to the redrive-to-queue function

Also fixed up the query for filling missing datasets. Nobody is going to SQL inject using product names.

### CODE REVIEW

File: datacube_alchemist/worker.py [hunk 1]
```diff
@@ -257,16 +257,16 @@ def find_fill_missing(self, queue, dryrun):
                 on sub_d.id = sub_s.dataset_ref
                 join agdc.dataset_type as sub_t
                 on sub_d.dataset_type_ref = sub_t.id
-                where sub_t.name = %{output_product}s
+                where sub_t.name = '{output_product}'
```
Comment (omad) [inline datacube_alchemist/worker.py:260]: How was this failing?

I prefer leaving correct escaping up to someone else's code, it's too easy to introduce silly bugs, even ignoring SQL injection vulns.

File: datacube_alchemist/cli.py [hunk 2]
```diff
@@ -212,16 +213,27 @@ def redrive_to_queue(queue, to_queue, limit):
...
-            message.delete()
-            count += 1
-            if limit and count >= limit:
-                break
-        else:
-            _LOG.error(f"Unable to send message {message} to queue")
-    _LOG.info(f"Completed sending {count} messages to the queue")
+    count_messages = dead_queue.attributes.get("ApproximateNumberOfMessages")
+
+    if count_messages == 0:
```
Comment (omad) [inline datacube_alchemist/cli.py:218]: It's not worth trying anyway? It isn't called 'ApproximateNumberOfMessage' for nothing. :)

**Review by omad — COMMENTED** [review]
Curious about the SQL...

File: datacube_alchemist/cli.py (hunk 2 above)
Comment (alexgleith) [reply datacube_alchemist/cli.py:218]: Maybe you're right. Doesn't make much difference.

File: datacube_alchemist/worker.py (hunk 1 above)
Comment (alexgleith) [reply datacube_alchemist/worker.py:260]: I'm not sure how to pass in variables properly. 

The error was:
```
Traceback (most recent call last):
  File "/env/bin/datacube-alchemist", line 8, in <module>
    sys.exit(cli_with_envvar_handling())
  File "/env/lib/python3.6/site-packages/datacube_alchemist/cli.py", line 66, in cli_with_envvar_handling
    cli(auto_envvar_prefix="ALCHEMIST")
  File "/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/env/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/env/lib/python3.6/site-packages/datacube_alchemist/cli.py", line 191, in add_missing_to_queue
    n_messages = alchemist.find_fill_missing(queue, dryrun)
  File "/env/lib/python3.6/site-packages/datacube_alchemist/worker.py", line 300, in find_fill_missing
    query, dict(input_products=input_products, output_product=output_product)
TypeError: 'dict' object does not support indexing
```

**Review by omad — APPROVED** [review]

File: datacube_alchemist/cli.py (hunk 2 above)
Comment (omad) [reply datacube_alchemist/cli.py:218]: Agreed :)

**Pull request merged**